elements/pl-string-input/pl-string-input.py

```diff
@@ -140,7 +145,7 @@ def parse(element_html, data):
     element = lxml.html.fragment_fromstring(element_html)
     name = pl.get_string_attrib(element, 'answers-name')
     # Get allow-blank option
-    allow_blank = pl.get_string_attrib(element, 'allow-blank', False)
+    allow_blank = pl.get_boolean_attrib(element, 'allow-blank', False)
```

Comment:
It might be a good idea to revisit the `allow_blank` option. The code here is problematic:
https://github.com/PrairieLearn/PrairieLearn/blob/2df88e38ed39c4026253afe8ae639d7a76ec5ee2/elements/pl-string-input/pl-string-input.py#L157

Consider two different student actions:
1) Open question, click grade.
2) Open question, enter text, delete all of this text, click grade.

In both cases, the input box will be blank, i.e., empty.

If there is a difference between these cases, then it's confusing for the user, because they can't tell the difference between empty and empty-after-deleting-text.

If there is no difference between these cases, then we will never arrive at the `if not a_sub and not allow_blank` line, because we will have found that `a_sub is None`.

Reply:
@tbretl: In talking with @mfsilva22 in office hours, I think it wasn't entirely clear what the issue was here. Is it that we should be treating a null or missing value the same as a zero-length string value, but we are failing to do that?

Reply:
I never really understood the need of the allow_blank option... I cant imagine a situation where it will be needed. So initially I thought it should be just considered invalid, but I was asked to implement this option. So if we are to keep the option, can you please suggest how you would like to address the issue above?

Reply:
@mfsilva22 Sorry for the confusion!

If I understand correctly, `allow_blank` means that the empty string is an acceptable answer. One reasonable way for a student to input the empty string is to input nothing - that is, to leave the input box blank and just click `Grade`. When the student does this, `data[‘submitted_answers’][name]` will have the value `None`, and the submission will be declared invalid.

If that makes sense (?), then I would do one of three things:

1. Eliminate the `allow_blank` option.
2. Change `if a_sub is None` to `if a_sub is None and not allow_blank`.
3. Add a button (e.g., in the input group between the input box and the question mark) that says something like “submit empty string” or “blank” and that, if pressed, will disable the input box and will (in `parse`) cause the submitted answer to be interpreted as blank.

I suggest option 1. I would also be on board with something like option 3.

I don’t like option 2, since it conflicts with our general philosophy of trying to avoid “I accidentally clicked `Grade`” situations.

Reply:
I'm fine with 1.